Modules/SubModule_Import/xASL_imp_NII2BIDS_Subject.m

```diff
@@ -16,68 +16,62 @@ function xASL_imp_NII2BIDS_Subject(imPar, bidsPar, studyPar, nameSubject)
+    end
+    
+    %% 2. Process the anat & perfusion files
+    listSessions = xASL_adm_GetFileList(fullfile(imPar.TempRoot,nameSubjectSession),'^ASL.+$',false,[],true);
+    
+    % Go through all sessions
+    for iSession = 1:length(listSessions)
+        xASL_imp_NII2BIDS_Session(imPar, bidsPar, studyPar, listSessions, nameSubjectSession, bidsLabel, iSession);
+    end
+    
+end
 
-	% Make a subject directory
-	if ~exist(fullfile(imPar.BidsRoot,['sub-' subjectLabel]),'dir')
-		mkdir(fullfile(imPar.BidsRoot,['sub-' subjectLabel]));
-	end
 
-    %% 2. Process all the anatomical files
-    % Go throught the list of anat files
-    for iAnatType = bidsPar.listAnatTypes
 
-        % Check if it exists
-        anatPath = '';
-        if xASL_exist(fullfile(imPar.TempRoot,nameSubject,[iAnatType{1},'.nii']),'file')
-            anatPath = fullfile(imPar.TempRoot,nameSubject,iAnatType{1});
-        end
+%% Check if there is a visit alias within the subject/visit name
+function bidsLabel = xASL_imp_CheckForAliasInVisit(imPar,nameSubjectSession)
 
-        if xASL_exist(fullfile(imPar.TempRoot,nameSubject,[iAnatType{1} '_1'],[iAnatType{1},'.nii']),'file')
-            anatPath = fullfile(imPar.TempRoot,nameSubject,[iAnatType{1} '_1'],iAnatType{1});
-        end
-
-        % If anatomical file of this type exist, then BIDSify its structure
-        if ~isempty(anatPath)
-
-            % Create the anatomical directory
-            if ~exist(fullfile(imPar.BidsRoot,['sub-' subjectLabel],'anat'),'dir')
-                mkdir(fullfile(imPar.BidsRoot,['sub-' subjectLabel],'anat'));
-            end
+    % Get visitAliases from imPar
+    visitAliases = imPar.tokenVisitAliases(:,2);
+    
+    % Separator subject/visit
+    separator = '_';
 
-            % Move the NiFTI file
-            xASL_Move([anatPath '.nii'],fullfile(imPar.BidsRoot,['sub-' subjectLabel],'anat',...
-                ['sub-' subjectLabel '_' iAnatType{1} '.nii.gz']),1);
+    % Default
+    subjectName = nameSubjectSession;
+    visitName = '';
 
-            % Load the JSON
-            jsonAnat = spm_jsonread([anatPath,'.json']);
-			
-            % Save the JSON
-			jsonAnat = xASL_bids_BIDSifyAnatJSON(jsonAnat,studyPar);
-            jsonAnat = xASL_bids_VendorFieldCheck(jsonAnat);
-            jsonAnat = xASL_bids_JsonCheck(jsonAnat,'');
-			
-            spm_jsonwrite(fullfile(imPar.BidsRoot,['sub-' subjectLabel],'anat',['sub-' subjectLabel '_' iAnatType{1} '.json']),jsonAnat);
+    % Iterate over aliases
+    for iAlias = 1:numel(visitAliases)
+        checkExpression = regexp(nameSubjectSession, [separator visitAliases{iAlias,1} '$'], 'once');
+        if ~isempty(checkExpression) % nameSubject should end in the visit alias
+            visitName = nameSubjectSession(checkExpression:end);
+            subjectName = nameSubjectSession(1:checkExpression-1);
         end
     end
+    
+    bidsLabel.subject = xASL_adm_CorrectName(subjectName,2);
```

Comment:
Are we sure here that we correct only the 'value' (for the keys `sub-` and `ses-`)? Or should this be done somewhere else to ensure that this is always the value?

Please also add a bit more comments, it's quite difficult to follow these scripts. I know that you're in a rush but this will backfire.

Reply:
Like with a few other comments I have to say that this wasn't necessarily created in a rush. This was like this before and just moved. Before it was called `nameSubject`, but other than that we used the same correction function and everything. Adding more comments to code in general can be helpful though 😉 
Regarding the specific functionality: I had to figure this out first too, but a simple debug made it super clear. We extract subject/visit from the name, so you could have something like `sub-001_ses-1` or whatever and there we basically only want the 001 and the 1. We need to get rid of the hyphens and underscores, which is why we use xASL_adm_CorrectName. We did this exactly like this before though and was probably originally written by @jan-petr.